iocore/net/UnixNetVConnection.cc

```diff
@@ -1456,7 +1456,7 @@ UnixNetVConnection::add_to_keep_alive_queue()
   if (lock.is_locked()) {
     nh->add_to_keep_alive_queue(this);
   } else {
-    ink_release_assert(!"BUG: It must have acquired the NetHandler's lock before doing anything on keep_alive_queue.");
+    Error("BUG: It must have acquired the NetHandler's lock before doing anything on keep_alive_queue.");
```

Comment:
I’d prefer ink_assert.

Reply:
it's not an assert situation as HttpSM is not always on original thread.

Reply:
But u must hold nh lock right ? Otherwise the queue will be broken by multi-access .

Reply:
And I don't see any reschedule and try to grab lock again logic  to protect this queue.

Reply:
unless there is a recent change that i missed, shouldn't not adding the queue close the connection? why will there be connection leaking?  If it is not going to close the connection, i would prefer to reschedule instead of ink_assert as it's going to leak connections in release.

Reply:
Maybe I miss something ? But I don't found the checking whether vc is in active queue or inactive queue in current logic. And when we insert failed it has no flag to signal SM that insertion has failed. So I think that will leak some connections in client keep-alive cases. Because SM assume the `xx_add_queue`  always success.

Or I missed something ?

Reply:
I started this PR as a quick work around to avoid the assert crashes in production. After this discussion,  i am not certain this PR wouldn't cause connection leaks. I will close the PR for now.

Reply:
Talk with @oknet offline. He said that the connection will not leak because it managed by cop_list and inactive_timeout is still running. But will leaking with inactive_timeout = 0, since the connections does not insert into keep_alive_queue.  ATS can not close connection directly, which is in keepalive queue, when ats's connections is overload. That will cause ats refuse to accept.